Make right-clicking on unfocused scene tabs possible

vaner-org opened the pull request:
Should be merged only after #112904.

This PR allows the right-click menu on a scene's tab to open without the editor also switching to that scene.

Additionally, the following menu options have been adapted to accept the clicked scene as context:
- Save Scene
- Save Scene As
- Show in FileSystem
- Play This Scene
- Set as Main Scene
- Close Tab
- Close Other Tabs
- Close Tabs to the Right

This should help tab management be more frictionless when dealing with complex scenes, especially since the close button only ever appears for the currently focused scene tab.

KoBeWi reviewed (commented):
It feels very brittle that the scene tabs will set a property of EditorNode and then the editor acts based on that property, and it has to be manually cleared. It's rather bug-prone.

Are there cases when `tab_context_idx` is needed after the menu callback? You could add a method to EditorSceneTabs that return the tab for which the option should be applied, but then it would be cleaner if it was used only for menu options from scene tabs (which means the scene tabs would have to get its own option id for scene save etc.), but without duplicating too much the code.

So for example the code for closing scene could look like
```C++
case SCENE_CLOSE: {
	_scene_tab_closed(editor_data.get_edited_scene());
} break;
case EditorSceneTabs::SCENE_CLOSE {
	int scene_to_close = scene_tabs->get_option_tab();
	_scene_tab_closed(scene_to_close);
} break;
```

vaner-org commented:
Okay, the implementation itself should be free of redundancies now, barring the second query of `get_option_tab` in `_dialog_action`. But since the file dialog takes focus away from the editor, I don't think that would cause problems.

As for the new enum names and their new placements, I would appreciate directed feedback. Not sure what should be `SCENE_`, what should be `SCENE_TAB_`, and where things belong.

vaner-org commented:
I've rewritten and retested this, should be a lot more stable.

> Are there cases when tab_context_idx is needed after the menu callback?

 `tab_closing_idx` had been replaced by `tab_context_idx`, which I've reverted, and most of the current functionality is left untouched. New cases have been introduced as `EditorSceneTabs::` and `SCENE_TAB` enums.

vaner-org committed:
Right-click on scene tab without selecting it
